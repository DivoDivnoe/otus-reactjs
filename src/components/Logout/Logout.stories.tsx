import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Logout } from './Logout';

export default {
  title: 'Example/Logout',
  component: Logout,
  argTypes: {
    logout: { action: 'logout' },
  },
} as ComponentMeta<typeof Logout>;

const Template: ComponentStory<typeof Logout> = (args) => <Logout {...args} />;

export const Default = Template;
