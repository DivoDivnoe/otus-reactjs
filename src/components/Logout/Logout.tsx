import React, { FC } from 'react';
import styled from '@emotion/styled';

const LogoutItem = styled.span`
  cursor: pointer;

  &:hover {
    color: #058aba;
  }
`;

export interface LogoutProps {
  logout: () => void;
}

export const Logout: FC<LogoutProps> = ({ logout }) => {
  return <LogoutItem onClick={logout}>Выйти</LogoutItem>;
};
