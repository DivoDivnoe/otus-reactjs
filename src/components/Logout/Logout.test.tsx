import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Logout, LogoutProps } from './Logout';

describe('Logout', () => {
  it('renders correctly', () => {
    const mocks: LogoutProps = {
      logout: jest.fn(),
    };

    const { asFragment } = render(<Logout {...mocks} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles click event correctly', () => {
    const mocks: LogoutProps = {
      logout: jest.fn(),
    };
    render(<Logout {...mocks} />);

    fireEvent.click(screen.getByText('Выйти'));
    expect(mocks.logout).toHaveBeenCalledTimes(1);
  });
});
