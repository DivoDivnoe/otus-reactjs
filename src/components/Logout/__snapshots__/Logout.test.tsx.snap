// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<DocumentFragment>
  <span
    class="css-1b0a5un"
  >
    Выйти
  </span>
</DocumentFragment>
`;
